psydac/api/tests/test_api_2d_compatible_spaces.py

```diff
@@ -86,95 +91,126 @@ def run_system_1_2d_dir(f0, sol, ncells, degree):
     return l2_error
 
 #==============================================================================
-def run_system_2_2d_dir(f1, f2,u1, u2, ncells, degree):
-    # ... abstract model
-    domain = Square()
+def run_stokes_2d_dir(domain, f, ue, pe, *, homogeneous, ncells, degree, scipy=False):
 
+    # ... abstract model
     V1 = VectorFunctionSpace('V1', domain, kind='H1')
     V2 = ScalarFunctionSpace('V2', domain, kind='L2')
     X  = ProductSpace(V1, V2)
 
-    x,y = domain.coordinates
-
-    F = element_of(V1, name='F')
-
-    u,v = [element_of(V1, name=i) for i in ['u', 'v']]
-    p,q = [element_of(V2, name=i) for i in ['p', 'q']]
+    u, v = elements_of(V1, names='u, v')
+    p, q = elements_of(V2, names='p, q')
 
+    x, y  = domain.coordinates
     int_0 = lambda expr: integral(domain , expr)
 
-    
-    a  = BilinearForm(((u,p),(v,q)), int_0(inner(grad(u),grad(v)) + div(u)*q - p*div(v)) )
-    l  = LinearForm((v,q), int_0(f1*v[0]+f2*v[1]+q))
-    
-    bc = EssentialBC(u, 0, domain.boundary)
-    equation = find([u,p], forall=[v,q], lhs=a((u,p),(v,q)), rhs=l(v,q), bc=bc)
+    a  = BilinearForm(((u, p), (v, q)), int_0(inner(grad(u), grad(v)) - div(u)*q - p*div(v)) )
+    l  = LinearForm((v, q), int_0(dot(f, v)))
 
-    error = Matrix([F[0]-u1, F[1]-u2])
-    l2norm_F = Norm(error, domain, kind='l2')
+    if homogeneous:
+        bc = EssentialBC(u, 0, domain.boundary)
+    else:
+        bc = EssentialBC(u, ue, domain.boundary)
+
```

Comment:
for documentation purposes we could add a remark specifying that the argument could be a function only defined on the boundary (instead of ue which is not known in a usual problem)

Reply:
Indeed the argument is a symbolic expression `g(x, y)`, which will only be evaluated at the boundary. Could this expression be valid only at the boundary? It is hard to see how such an expression could be constructed, given that it is symbolic.

@saidctb Do you remember if we accept a generic symbol for `ue`, which we can then substitute for a field?

Reply:
we never tested that, but it should be possible

Reply:
Sorry, my remark was not very clear. I meant: "specifying that the argument could be a function which only represents the boundary data". This boundary data can be defined outside of the boundary, but there it will not correspond to the exact solution.

Reply:
I've added a comment with commit caf7b5f